src/seagap/harmonic_mean.py

```diff
@@ -0,0 +1,121 @@
+import math
+from typing import Union
+
+import numba
+import numpy as np
+import pandas as pd
+
+
+@numba.njit
+def _compute_hm(
+    dd: np.ndarray,
+    sv: np.ndarray,
+    start_depth: Union[int, float],
```

Comment:
wouldn't `start_depth: float` be sufficient since that can also take an integer argument?